modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/DefaultMetadataScopeAdapter.java

```diff
@@ -60,32 +48,32 @@
  */
 public final class DefaultMetadataScopeAdapter implements MetadataScopeAdapter {
 
-  private Supplier<NullMetadataResolver> nullMetadataResolverSupplier = new NullMetadataResolverSupplier();
-  private Supplier<? extends TypeKeysResolver> keysResolver = nullMetadataResolverSupplier;
-  private Map<String, Supplier<? extends InputTypeResolver>> inputResolvers = new HashMap<>();
+  private JavaKeyIdResolverModelParser javaKeyIdResolverModelParser =
+      new JavaKeyIdResolverModelParser(null, null, org.mule.sdk.api.metadata.NullMetadataResolver.class);
```

Comment:
Create a NullKeyIdResolverModelParser, which receives no arguments in the constructor and directly resolves to the correct values. This also goes for the other model parsers.